pkg/config/hooks.go

```diff
@@ -247,93 +247,93 @@ func stringToByteArrayHook(f reflect.Type, t reflect.Type, data interface{}) (in
 }
 
 func stringToTimeDurationPointerHook(f reflect.Type, t reflect.Type, data interface{}) (interface{}, error) {
-	if f != nil && f.Kind() != reflect.String {
+	d := time.Duration(0)
+	if f.Kind() != reflect.String || t != reflect.TypeOf(&d) {
 		return data, nil
 	}
-	if s, ok := data.(string); ok {
-		d := time.Duration(1)
-		if t == reflect.TypeOf(&d) {
-			if s == "" {
-				return nil, nil
```

Comment:
This untyped `nil` causes future hooks to have a `nil` `f` / from type. This is invalid - if really the empty string means `nil`, it has to be a _typed_ `nil`. Also since `f.Kind() == reflect.String` we can (and should) assume that `data` is a string.

Same for the next hooks.